File: valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        paranthesis = {'{':'}','[':']','(':')'}
        stack = []
        for letter in s:
            if letter in paranthesis.keys():
                stack.append(paranthesis[letter])
            elif letter in stack and stack[-1]==letter:
                stack.pop()
            else:
                return False
        return True  if len(stack) == 0 else False

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();

        for(char ch : s.toCharArray()) {

            if(ch == '(' || ch == '{' || ch == '[') {
                st.add(ch);
            } else {
                if(st.isEmpty()) {
                    return false;
                }

                char stCh = st.pop();
                if(!check(stCh, ch)) {
                    return false;
                }

            }
        }

        if(st.size() > 0) {
            return false;
        }

        return true;
    }
        
    public boolean check(char stCh, char ch) {

        if((stCh == '(' && ch == ')') ||
           (stCh == '{' && ch == '}') ||
           (stCh == '[' && ch == ']')) {
            return true;
        }

        return false;

    }
}
